docs/dev/Kibana-Reporting-Design-Proposal.md

```diff
@@ -20,43 +21,111 @@ Kibana Reports for Open Distro allows ‘Report Owner’ (engineers, including b
+1. As a user, when a report definition has been triggered via schedule or alert, I should be able to see an instance of the report in my "Reports" list, and any users I send the report to, will also see a report instance in their repective "Reports" list page.
+1. As a user, when a report is created from a triggered definition condition, I should receive a notification on all configured delivery channels (Email, chime, slack, etc)
+1. As a user, when a report is created from a triggered definition condition, I should receive a kibana toast alert letting me know that the report is available.
+1. As a user, when a report fails due to an error, I should receive a kibana toast alert letting me know that the report failed.
+1. As a user, when a report fails due to an error, I should see the report with a failed status on the `Reports` list.
+
+#### View report definition details
+1. As a user, I should be able to view all my saved definition settings.
+1. As a user, I should be able to edit, duplicate or delete a report definition.
+    1. As a user, when I delete a report definition, any schedule or alert based triggers will automatically stop.
+
+#### Edit report definition
+1. As a user, I should be able to edit the report settings, triggers and delivery settings.
+1. As a user, I should be able to save my updated report definition.
+
+#### View report details
+1. As a user, I should be able to see both created date, and last updated date.
+1. As a user, I should be able to share an existing report.
+1. As a user, I should be able to archive an existing report. (p2)
+1. As a user, I should be able to downlaod available report files (PDF, PNG, CSV, Excel, etc)
+1. As a user, I should be able to copy the permalink to a report source.
+
+##### Sharing or archiving an existing report
+1. As a report owner, I should be able to share or archive a report.
+    1. As report owner, I should be able to add new email recepients, and new kibana recepients to an existing report.
+    1. As a report owner, I sholud not be able to remove existing email or kibana recepients from an existing report.
+    1. As a report owner, I should receive a Kibana toast notification when a report has been delivered to new recepients.
+    1. As a report recepient, I should not get new notifications or emails when additional users have been added to an existing report.
+1. As a report recepient, I can only archive a report.
+
+
+### Functional
 * The user should be able to create/modify reports both through the Kibana UI or programmatically through APIs.
 * Ability to schedule/trigger reports periodically (a cron based schedule) or at a given frequency continuously or within certain time range.
 * Ability on instantly create downloadable reports or deliver scheduled reports via external channels.
 * Ability to enable/disable report generation
+* Ability to generate download logs as part of fine-grained audit logs.
```

Comment:
Need clarity on what is meant by logs? What kind of logs?

Reply:
Application logs via FGAC audit logs. See documentation in Security plug-in (https://opendistro.github.io/for-elasticsearch-docs/docs/security-audit-logs/) Depending on the level of logging enabled, download actions should be part of the generated logs. This is more of a backend question than front-end.